extension/observer/hostobserver/extension.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hostobserver
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"syscall"
+
+	"github.com/shirou/gopsutil/net"
+	"github.com/shirou/gopsutil/process"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+type hostObserver struct {
+	observer.EndpointsWatcher
+	logger *zap.Logger
+}
+
+var _ component.ServiceExtension = (*hostObserver)(nil)
+
+func newObserver(logger *zap.Logger, config *Config) (component.ServiceExtension, error) {
+	h := &hostObserver{
+		logger: logger,
+		EndpointsWatcher: observer.EndpointsWatcher{
+			RefreshInterval: config.RefreshInterval,
+			ListEndpoints:   (&hostObserver{}).listEndpoints,
```

Comment:
This is pretty ugly creating a completely separate instance of the host observer to pass the listEndpoints method to.  Why don't you just make listEndpoints a plain function instead?

Reply:
Also logger would be `nil` in listEndpoints which is not going to behave properly.

Reply:
Will refactor this and also make sure there's test coverage around this!